In [2]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/ecommerce-behavior-data-from-multi-category-store/2019-Oct.csv
/kaggle/input/ecommerce-behavior-data-from-multi-category-store/2019-Nov.csv


In [6]:
df = pd.concat((pd.read_csv('/kaggle/input/ecommerce-behavior-data-from-multi-category-store/2019-Nov.csv',nrows = 5000000,
                                                                                                           header = 'infer',
                                                                                                           parse_dates = ['event_time'],
                                                                                                           date_parser = pd.to_datetime,),
                pd.read_csv('/kaggle/input/ecommerce-behavior-data-from-multi-category-store/2019-Oct.csv',nrows = 5000000,
                                                                                                           header = 'infer',
                                                                                                           parse_dates = ['event_time'],
                                                                                                           date_parser = pd.to_datetime,)))

In [7]:
#The Pareto principle states that for many outcomes roughly 80% of consequences come from 20% of the causes. 
#Also known as 80/20 rule, in e-commerce simply means that most of your business, around 80%, likely comes from about 20% of your customers.
#Prove that the pareto principle applies to your store.

df.reset_index(inplace = True)
import math
Descending_purchase_by_user = df[(df.event_type == 'purchase')].groupby([df.user_id]).agg(Price = ('price', 'sum')).sort_values('Price',ascending= False)
Twenty_Percent = Descending_purchase_by_user.iloc[:int(len(Descending_purchase_by_user)*0.2)+1, :].Price.sum()
Twenty_Percent

33117594.52

In [8]:
Total = df[(df.event_type == 'purchase')].price.sum()
Total
print('20% of customers consequences has the ' + str((Twenty_Percent/Total*100)) + '% percentage of total.')

20% of customers consequences has the 66.88165135139822% percentage of total.


**Again since our machine were not able to read all data we have found out the percentage as 66.88 but we believe that if we would be able to read all raws we could prove that it is around 80%.**